backend/.husky/pre-commit

```diff
@@ -0,0 +1,3 @@
+cd backend
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/1#issuecomment-2348621052>#3fc86c</a><span> - 09/13/2024, 10:46 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The pre-commit hook script lacks error handling and may continue execution even if a command fails
</div>
<div id="fix">
<b>Fix:</b> Add error handling to the pre-commit hook script by using set -e to exit on any command failure and wrapping the commands in a function with proper error messaging
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -1,3 +1,12 @@
 +#!/bin/sh
 +set -e
 +
 +run_checks() {
  cd backend
  npm run lint:fix
  npm run format
 +}
 +
 +if ! run_checks; then
 +  echo "Pre-commit hook failed. Please fix the errors and try again."
 +  exit 1
 +fi
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged

Reply:
will update this

Reply:
updated